src/csharp/Microsoft.Spark/Interop/Ipc/JvmBridge.cs

```diff
@@ -83,8 +88,9 @@ private ISocketWrapper GetConnection()
             _socketSemaphore.Wait();
             if (!_sockets.TryDequeue(out ISocketWrapper socket))
             {
+                IPEndPoint dotnetBackendIPEndpoint = SparkEnvironment.ConfigurationService.GetBackendIPEndpoint();
                 socket = SocketFactory.CreateSocket();
```

Comment:
Pretty sure this is a bug. It's not using the backend ip endpoint, instead using the default constructor which favors the default IP, causing the PR to break (it's using 127.0.0.1 no matter what).